From 9cce8d06b09a945f7eeb512e0b397680a01051fe Mon Sep 17 00:00:00 2001
From: Thierry Reding <thierry.reding@avionic-design.de>
Date: Tue, 25 Sep 2012 15:21:30 +0200
Subject: [PATCH] Allow multitouch support to be disabled

Instead of always automatically including multitouch support if the
mtdev library is installed, the new --disable-multitouch option can
be used to forcefully disable multitouch support in the driver.

Signed-off-by: Thierry Reding <thierry.reding@avionic-design.de>
---
 configure.ac | 19 +++++++++++++------
 1 file changed, 13 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index dff2bcf..ed80ca1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -50,12 +50,19 @@ PKG_CHECK_MODULES(UDEV, libudev)
 
 PKG_CHECK_MODULES(XI22, [inputproto >= 2.1.99.3] [xorg-server >= 1.11.99.901], HAVE_XI22="yes", HAVE_XI22="no")
 
-if test "x$HAVE_XI22" = xyes; then
-        # Obtain compiler/linker options for mtdev
-        PKG_CHECK_MODULES(MTDEV, mtdev, HAVE_MTDEV="yes", HAVE_MTDEV="no")
-fi
-if test "x$HAVE_XI22" = xyes && test "x$HAVE_MTDEV" = xyes; then
-        AC_DEFINE(MULTITOUCH, 1, [XI2.2 available])
+AC_ARG_ENABLE(multitouch,
+              AS_HELP_STRING([--disable-multitouch],
+                             [Disable multitouch support (default: no)]),
+              [], [enable_multitouch=yes])
+
+if test "x$enable_multitouch" != "xno"; then
+        if test "x$HAVE_XI22" = xyes; then
+                # Obtain compiler/linker options for mtdev
+                PKG_CHECK_MODULES(MTDEV, mtdev, HAVE_MTDEV="yes", HAVE_MTDEV="no")
+        fi
+        if test "x$HAVE_XI22" = xyes && test "x$HAVE_MTDEV" = xyes; then
+                AC_DEFINE(MULTITOUCH, 1, [XI2.2 available])
+        fi
 fi
 
 # Define a configure option for an alternate input module directory
-- 
1.7.12.1

